### PR TITLE
III-5234 - Make titles dynamic based on scope

### PR DESCRIPTION
**Pull request by brampauwelyn**
### Changed
- Make titles dynamic based on scope
Screenshot 2023-01-19 at 14 57 51

---
Ticket: https://jira.uitdatabank.be/browse/III-5234

### CODE REVIEW

File: src/i18n/fr.json [hunk 1]
```diff
@@ -110,7 +110,10 @@
       "title": "Que voulez-vous ajouter"
     },
     "calendar": {
-      "title": "Quand cet événement ou cette activité a-t-il lieu ?",
+      "title": {
+        "events": "Quand cet événement ou cette activité a-t-il lieu ?",
+        "places": "Quand cet endroit ou cet emplacement est-il ouvert ?"
```
Comment (simon-debruijn) [inline src/i18n/fr.json:115]:
```suggestion
        "events": "Quand cet événement ou cette activité a-t-il lieu?",
        "places": "Quand cet endroit ou cet emplacement est-il ouvert?"
```

File: src/pages/steps/CalendarStep/CalendarStep.tsx [hunk 2]
```diff
@@ -299,7 +299,7 @@ const CalendarStep = ({ offerId, control, ...props }: CalendarStepProps) => {
 const calendarStepConfiguration: StepsConfiguration = {
   Component: CalendarStep,
   name: 'calendar',
-  title: ({ t }) => t('create.calendar.title'),
+  title: ({ t }) => t('create.calendar.title.events'),
```
Comment (simon-debruijn) [inline src/pages/steps/CalendarStep/CalendarStep.tsx:302]:
```suggestion
  title: ({ t, scope }) => t(`create.calendar.title.${scope}`),
```

File: src/i18n/fr.json (hunk 1 above)
Comment (brampauwelyn) [reply src/i18n/fr.json:115]: In French, exclamation points, question marks, colons, and semi-colons (all forms of "high" punctuation(Opens in a new window)) should always have a space preceding them.

File: src/pages/steps/LocationStep.tsx [hunk 3]
```diff
@@ -381,7 +381,7 @@ const locationStepConfiguration: StepsConfiguration = {
   Component: LocationStep,
   name: 'location',
   shouldShowStep: ({ watch }) => !!watch('typeAndTheme')?.type?.id,
-  title: ({ t }) => t('create.location.title'),
+  title: ({ t }) => t('create.location.title.events'),
```
Comment (simon-debruijn) [inline src/pages/steps/LocationStep.tsx:384]:
```suggestion
  title: ({ t, scope }) => t(`create.location.title.${scope}`),
```

File: src/pages/create/OfferForm.tsx [hunk 4]
```diff
@@ -259,8 +259,14 @@ const OfferForm = () => {
...
-        calendarStepConfiguration,
-        locationStepConfiguration,
+        {
+          ...calendarStepConfiguration,
+          title: () => t(`create.calendar.title.${scope}`),
+        },
+        {
+          ...locationStepConfiguration,
+          title: () => t(`create.location.title.${scope}`),
+        },
```
Comment (simon-debruijn) [inline src/pages/create/OfferForm.tsx:269]: This can be reverted if you can use scope in the configurations below

File: src/i18n/fr.json (hunk 1 above)
Comment (simon-debruijn) [reply src/i18n/fr.json:115]: Wow 😲 Didn't know that

File: src/pages/create/OfferForm.tsx (hunk 4 above)
Comment (brampauwelyn) [reply src/pages/create/OfferForm.tsx]: Yes correct, also did it for `additionalInformationStep`

**Review by simon-debruijn — APPROVED** [review]

**Pull request merged**